src/transformers/models/esm/modeling_tf_esm.py

```diff
@@ -1102,6 +1103,11 @@ def __init__(self, config):
 
         self.esm = TFEsmMainLayer(config, add_pooling_layer=False, name="esm")
         self.lm_head = TFEsmLMHead(config, name="lm_head")
+        if config.tie_word_embeddings:
+            # Ensure word embeddings are built so that we actually have something to tie
+            with tf.name_scope(os.path.join(self._name_scope(), "esm", "embeddings", "word_embeddings")):
```

Comment:
Cries in TensorFlow 😭

Reply:
I know, but it's the only option when we want to create a weight for a distant sublayer! TF only 'walks' the name scope hierarchy via the call stack when weights are built during `call()`; every other time you have to explicitly enter the `tf.name_scope()` you want.